bend_gcode.py

```diff
@@ -142,46 +98,89 @@ def createSplineLookupTable():
             else:
                 outputFile.write(currentLine)
-print("GCode bending finished!")
\ No newline at end of file
+
+    print("\nGCode bending finished!")
+    print(f"Processed {index} lines in {timedelta(seconds=perf_counter()-start)}")
+
+if __name__ == "__main__":
+    from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+
+    parser = ArgumentParser(formatter_class=ArgumentDefaultsHelpFormatter)
+    parser.add_argument("in_file", help="input file name (*.gcode)")
+    parser.add_argument("-o", "--out_file", help="output filename, default 'BENT_{in_file}'")
+    parser.add_argument("-x", "--x_values", default=(125, 95), nargs="*", type=float,
+                        help=("x values that define the spline, space-separated (e.g. '125 50 33')."
+                              " First should be in the center of your part."))
+    parser.add_argument("-z", "--z_values", default=(0, 140), nargs="*", type=float,
+                        help=("corresponding z values that define the spline, space-separated."
+                              " First should be 0 (e.g. '0 80 140')."))
+    parser.add_argument("-l", "--layer_height", default=0.3, type=float,
+                        help="layer height of the sliced gcode [mm].")
+    parser.add_argument("-a", "--warning_angle", default=30, type=float,
+                        help="Maximum angle [degrees] printable with your setup")
+    parser.add_argument("-b", "--bend_angle", default=-30, type=float,
+                        help="Angle [degrees] of the spline at the top point")
+    parser.add_argument("-d", "--discretization_length", default=0.01, type=float,
+                        help="Discretization length for the spline length lookup table")
+    parser.add_argument("-s", "--skip_plot", action="store_true",
+                        help="flag to skip plotting of the spline")
+    parser.add_argument("--xy_precision", type=int, default=4,
+                        help="Decimals of precision to round x/y values to.")
+    parser.add_argument("--z_precision", type=int, default=3,
+                        help="Decimals of precision to round z (height) values to.")
+    parser.add_argument("--e_precision", type=int, default=5,
+                        help="Decimals of precision to round extrusion amounts to.")
+    parser.add_argument("--printer_dims", nargs=2, default=(200, 200), type=float,
+                        help="printer width and height [mm], space-separated.")
+
+    args = parser.parse_args()
+
+    spline = Spline(args.x_values, args.z_values, args.discretization_length,
+                    args.bend_angle)
+    if not args.skip_plot:
+        print("Press Q to close the plot and continue")
+        spline.plot(printer_dims=args.printer_dims)
+
+    out_file = args.out_file or f"BENT_{args.in_file}"
+
+    main(args.in_file, out_file, spline, args.layer_height, args.warning_angle,
+         args.xy_precision, args.z_precision, args.e_precision)
```

Comment:
I would have put all of this in a `cli` function, but that's just a style preference.

Reply:
I don't see a particular benefit to that, given it's already in the script portion (so can't be imported), and is only 'called' once. I've never seen a `cli` function before, so am curious how it would be implemented. Are you thinking something like

```python3
def cli():
    parser = ArgumentParser()
    parser.add_argument(...)
    ...
    return parser.parse_args()
```
or would the parser be returned instead of the parsed arguments?

Given it's your preferred approach, are there benefits I'm missing?